crates/aranya-client/src/aqc/net.rs

```diff
@@ -234,12 +230,7 @@ impl AqcClient {
                 .await
                 .ok_or(AqcError::ServerConnectionTerminated)?;
             // Receive a PSK identity hint.
-            // TODO: Instead of receiving the PSK identity hint here, we should
-            // pull it directly from the connection.
-            let identity = server_state
-                .identity_rx
-                .try_recv()
-                .assume("identity received after accepting connection")?;
+            let identity = get_conn_identity(&mut conn)?;
```

Comment:
Would be neat if the method could be invoked on the object:
```suggestion
            let identity = conn.get_identity()?;
```
I suppose that would require making our own fork of `s2n-quic` or wrapping it in our own type which seems like too much effort.